nimbleclient/v1/restclient.py

```diff
@@ -365,3 +415,16 @@ def perform_bulk_resource_action(self, resource, action, **params):
 
         resp = self.post(f"{self._ENDPOINTS[resource]}/actions/{action}", **params)
         return resp['data'] if 'data' in resp else resp
+
+    def build_advanced_criteria(self, filter):
+        """Construct Advanced Criteria JSON object"""
+
+        data = {}
```

Comment:
No point in defining a dict only to immediately set its parameters afterwards, in fact you don't need any of these variables. Just do this:

```python
return {
    'data': {
        '_constructor': 'AdvancedCriteria',
        'operator': operator,
        'criteria': criteria,
    },
    'operationType': 'fetch'
}

Reply:
Agreed. Fixed accordingly